app/models/input_template_renderer.rb

```diff
@@ -0,0 +1,39 @@
+class InputTemplateRenderer
+  include Foreman::Renderer
+
+  attr_accessor :template, :host, :invocation, :error_message
+
+  def initialize(template, host = nil, invocation = nil)
+    @host = host
+    @invocation = invocation
+    @template = template
+  end
+
+  def render
+    render_safe(@template.template, [ :input ])
+  rescue => e
+    self.error_message = _('error during rendering: %s') % e.message if error_message.nil?
+    Rails.logger.debug e.to_s + "\n" + e.backtrace.join("\n")
+    return false
+  end
+
+  def preview
+    @preview = true
+    output = render
+    @preview = false
+    output
+  end
+
+  def input(name)
+    input = @template.template_inputs.where(:name => name.to_s).first || @template.template_inputs.detect { |i| i.name == name.to_s}
```

Comment:
in theory users could write templates like this `<%= snippet(1) %>` or `<%= snippet(:name) %>`, as rubyist I'd expect the later to work